### PR TITLE
Added support to modify bot behaviour

### PR DESCRIPTION
**Pull request by StuxSoftware**
In preparation to support automatic fetching of links I thought we could add a marker
that prevents the bot to parse the message.

This marker is automatically added to the reply so that the reply may not be parsed in any
way.

In addition it can be used to activate/deactivate experimental features of the bot.

Add ffnbot!ignore to ignore the comment entirely
Add ffnbot!noreformat to suppress automatic reformatting (that routinely fails).

Side Note: FFN-Story instances are not cached anymore.

### CODE REVIEW

**Comment by StuxSoftware** [comment]
I want to add some rationale to explain why I felt this merge neccessary and why I did it like I did:
The most important reason is to implement a flag to completely opt out of the fanfiction-bot for the single post, most importantly as #6 needs a way to prevent responding to itself.

However shortly after that I wanted to make sure, that everybody can choose to opt out of features that are error prone (like the metadata formatting that needs to be redone see commit 358887e112342c5de73e44a71d8bec7e2ee30ca3)

Additionally I wanted to have a flag that users recognize as such and is self-explaining (mostly)

The entire "context" that is a set of all flags, is passed to every function involved in generating the response beginning at the function `formulate_reply(comment_body: str) -> Optional[str]`
I use this function to test the bot

Also I think, we do not need to document all of them in the readme as some of them are for testing purposes only.

**Comment by tusing** [comment]
Thanks for the explanation. Why don't we need to cache story objects anymore?

EDIT: Also, not sure about the need for ffnbot!ignore, if we aren't going to implement #6. People didn't seem too happy about the concept of the bot automatically replying to comments with fanfiction links - in fact, some people (even though it was a small minority) were adamant that they would rather format comments in their own way. If people don't want to use the bot, they don't have to call it in their comment, I think.

TL;DR: Even though the ignore feature couldn't hurt, I'm not sure it's necessary if #6 isn't going to be implemented.
